cmd/invoke.go

```diff
@@ -135,8 +135,17 @@ func runInvoke(cmd *cobra.Command, args []string, newClient ClientFactory) (err
 		return
 	}
 
+	// Load the function
+	f, err := fn.NewFunction(cfg.Path)
+	if err != nil {
+		return
+	}
+	if !f.Initialized() {
+		return fmt.Errorf("'%v' does not contain an initialized function", cfg.Path)
+	}
+
 	// Client instance from env vars, flags, args and user prompts (if --confirm)
-	client, done := newClient(ClientConfig{Namespace: cfg.Namespace, Verbose: cfg.Verbose, InsecureSkipVerify: cfg.Insecure})
```

Comment:
Shouldn't it be: `Namespace: f.Namespace` instead of deletion?

But it really depends on what is answer to https://github.com/knative-sandbox/kn-plugin-func/pull/1258#issuecomment-1255212969

Reply:
is so then also test should be updated:
 ```diff
 diff --git a/cmd/invoke_test.go b/cmd/invoke_test.go
index e06d4292..bf20a3a0 100644
--- a/cmd/invoke_test.go
+++ b/cmd/invoke_test.go
@@ -23,7 +23,7 @@ func TestInvoke(t *testing.T) {
        var invoked int32
 
        // Create a test function to be invoked
-       if err := fn.New().Create(fn.Function{Runtime: "go", Root: root}); err != nil {
+       if err := fn.New().Create(fn.Function{Runtime: "go", Root: root, Namespace: "foobar"}); err != nil {
                t.Fatal(err)
        }
 
@@ -62,14 +62,24 @@ func TestInvoke(t *testing.T) {
        }
        t.Cleanup(job.Stop)
 
+       usedNs := ""
+       newClient := func(conf ClientConfig, opts ...fn.Option) (*fn.Client, func()) {
+               usedNs = conf.Namespace
+               return NewClient(conf, opts...)
+       }
+
        // Test that the invoke command invokes
-       cmd := NewInvokeCmd(NewClient)
+       cmd := NewInvokeCmd(newClient)
        cmd.SetArgs([]string{})
 
        if err := cmd.Execute(); err != nil {
                t.Fatal(err)
        }
 
+       if usedNs != "foobar" {
+               t.Errorf("bad ns used: %q", usedNs)
+       }
+
        if atomic.LoadInt32(&invoked) != 1 {
                t.Fatal("function was not invoked")
        }
```

Reply:
Right, the namespace is unnecessary because the Function's current namespace will be used.
For example, regardless of the user's currently active kubernetes namespace, a request to invoke a Function should always attempt to invoke the function where it actually is deployed.